src/order/complete_well_founded.lean

```diff
@@ -49,6 +49,61 @@ same `Sup`. -/
 def is_Sup_finite_compact : Prop :=
 ∀ (s : set α), ∃ (t : finset α), ↑t ⊆ s ∧ Sup s = t.sup id
 
+/-- An element k of a complete lattice is said to be compact if any set that Sups
+above k has a finite subset that Sups above k -/
+def is_compact_element {α : Type*} [complete_lattice α] (k : α) :=
+∀ s : set α, k ≤ Sup s → ∃ t : finset α, ↑t ⊆ s ∧ k ≤ t.sup id
+
+/-- An equivalent characterization of a compact element k is that any directed
+that Sups above k must get above k at some point in the set -/
+theorem is_compact_element_iff_directed_Sup_finite (k : α) :
+  is_compact_element k ↔ ∀ s : set α, s.nonempty → directed_on (≤) s → k ≤ Sup s → ∃ x : α, x ∈ s ∧ k ≤ x :=
```

Comment:
If I'm reading [this page](https://encyclopediaofmath.org/wiki/Compact_lattice_element) correctly, the RHS of your iff is the condition of being a finite element.

So should we introduce:
```lean
def is_finite_element {α : Type*} [complete_lattice α] (k : α) :=
∀ s : set α, s.nonempty → directed_on (≤) s → k ≤ Sup s → ∃ x : α, x ∈ s ∧ k ≤ x
```
above and then use that here?

Reply:
Ah yes, those concepts can be distinct, but are equivalent for complete lattices. Given that, I think it'd be nice to define finite element and show equivalence in the complete lattice case.